ide/coqide/dune

```diff
@@ -26,14 +26,29 @@
 (library
  (name coqide_gui)
  (wrapped false)
- (modules (:standard \ document idetop coqide_main default_bindings_src))
+ (modules (:standard \ document idetop coqide_main default_bindings_src gen_ide_specific))
+ (foreign_stubs
+  (language c)
+  (names coqide_os_stubs))
  (optional)
  (libraries coqide-server.protocol coqide-server.core lablgtk3-sourceview3))
 
+(executable
+ (name gen_ide_specific)
+ (libraries coq-core.config)
+ (modules gen_ide_specific))
+
 (rule
- (targets coqide_os_specific.ml)
- (deps (:in-file coqide_X11.ml.in)) ; TODO support others
- (action (copy# %{in-file} %{targets})))
+ (targets coqide_os_specific.ml coqide_os_stubs.c)
+ (deps
+  (:gen gen_ide_specific.exe)
+  coqide_X11.ml.in
+  coqide_X11.c.in
+  coqide_QUARTZ.ml.in
+  coqide_QUARTZ.c.in
+  coqide_WIN32.ml.in
+  coqide_WIN32.c.in)
+ (action (run %{gen} %{targets})))
```

Comment:
instead of copying the files by hand, couldn't the ocaml program just spit the gtk backend name and then use that value and the standard `copy` dune directive?

Reply:
If we use the os instead of graphics  name, there should be a dune variable for it

Reply:
Dune variables are derived from `ocamlc -config`, which doesn't cover this use case well; see the relevant upstream bugs such as https://github.com/ocaml/ocaml/issues/10613

Reply:
> instead of copying the files by hand, couldn't the ocaml program just spit the gtk backend name and then use that value and the standard `copy` dune directive?

I don't know.